app/src/main/java/org/breezyweather/sources/climweb/ClimWebService.kt

```diff
@@ -0,0 +1,251 @@
+        location: Location,
+        feature: SourceFeature,
+    ): Boolean {
+        return (
+            feature == SourceFeature.FEATURE_ALERT &&
+                location.countryCode != null &&
+                location.countryCode.equals(countryCode, ignoreCase = true)
+            ) ||
+            (
+                feature == SourceFeature.FEATURE_NORMALS &&
+                    location.countryCode != null &&
+                    location.countryCode.equals(countryCode, ignoreCase = true) &&
+                    !cityClimatePageId.isNullOrEmpty()
+                )
+    }
+
+    override val currentAttribution = null
+    override val airQualityAttribution = null
+    override val pollenAttribution = null
+    override val minutelyAttribution = null
+
+    override fun requestSecondaryWeather(
+        context: Context,
+        location: Location,
+        requestedFeatures: List<SourceFeature>,
+    ): Observable<SecondaryWeatherWrapper> {
+        val failedFeatures = mutableListOf<SourceFeature>()
+        val alerts = if (requestedFeatures.contains(SourceFeature.FEATURE_ALERT)) {
+            mApi.getAlerts().onErrorResumeNext {
+                failedFeatures.add(SourceFeature.FEATURE_ALERT)
+                Observable.just(ClimWebAlertsResult())
+            }
+        } else {
+            Observable.just(ClimWebAlertsResult())
+        }
+
+        val normals = if (requestedFeatures.contains(SourceFeature.FEATURE_NORMALS) &&
+            !cityClimatePageId.isNullOrEmpty()
+        ) {
+            val cityId = location.parameters.getOrElse(id) { null }?.getOrElse("cityId") { null }
+            if (cityId != null) {
+                mApi.getNormals(cityClimatePageId!!, cityId).onErrorResumeNext {
+                    // Silently fail here: not all forecast cities have normals
+                    Observable.just(emptyList())
+                }
+            } else {
+                // Fail here: There should be a city ID in the location parameters
+                failedFeatures.add(SourceFeature.FEATURE_NORMALS)
+                Observable.just(emptyList())
+            }
+        } else {
+            Observable.just(emptyList())
+        }
+
+        return Observable.zip(alerts, normals) {
+                alertsResult: ClimWebAlertsResult,
+                normalsResult: List<ClimWebNormals>,
+            ->
+            convert(
+                location = location,
+                source = alertAttribution,
+                alertsResult = alertsResult,
+                normalsResult = normalsResult,
+                failedFeatures = failedFeatures
+            )
+        }
+    }
+
+    // Location parameters
+    override fun needsLocationParametersRefresh(
+        location: Location,
+        coordinatesChanged: Boolean,
+        features: List<SourceFeature>,
+    ): Boolean {
+        // As a secondary source, this function does not receive any value for features.
```

Comment:
Fixed in 3789ad4c2d452ac857825a001c3d4d038653612e

Reply:
Oh nice! I will incorporate that change.